### PR TITLE
Update release check list

### LINKED ISSUE #10: Fix up the release checklist for CLI
Related to https://github.com/oras-project/oras/issues/256

### PR DESCRIPTION
**Pull request by qweeah**
Since releasing is update in oras-go/oras#377, this PR updated the checklist.

Resolves #10

### CODE REVIEW

File: docs/CLI/5_developer_guide.md [hunk 1]
```diff
@@ -105,22 +109,19 @@ To update or add new dependencies, run `go get <package name>`.
     Git commit:     173c010570c48e4aa18ce186cae8cc812f8e8b6e
     Git tree state: clean
     ```
-14. Create a PR on [fishworks/fish-food](https://github.com/fishworks/fish-food) with the new ORAS version and checksums. This can be done by navigating to [this file](https://github.com/fishworks/fish-food/blob/main/Food/oras.lua) in your browser, and clicking the "Edit" button, making the changes and clicking "Propose changes", which will allow you to open a PR. You can also do this in your terminal/editor, of course.
```
Comment (qweeah) [inline docs/CLI/5_developer_guide.md:108]: Removed since the tool is archived, see https://github.com/fishworks/fish-food#this-project-is-being-archived

**Comment by vsoch** [comment]
See discussion in #26 for a few things to try! I do think the person that has access to Netlify needs to try these things out. But actually @qweeah I think you could try tweaking https://github.com/oras-project/oras-www/blob/main/netlify.toml too :)

**Comment by qweeah** [comment]
> See discussion in #26 for a few things to try! I do think the person that has access to Netlify needs to try these things out. But actually @qweeah I think you could try tweaking https://github.com/oras-project/oras-www/blob/main/netlify.toml too :)

Thanks for the advice @vsoch. Is https://github.com/oras-project/oras-www/pull/28 supposed to fix the solution of the build error?

**Comment by vsoch** [comment]
@qweeah yes in #28 I do the same build (but with my fixes) and you can see it's successful! So we just need a maintainer to review that.

**Review by shizhMSFT — COMMENTED** [review]
Could you re-number the steps?

**Comment by qweeah** [comment]
> Could you re-number the steps?

Looks like markdown will fix the gaps in sequence number. Still will do.

File: docs/CLI/5_developer_guide.md [hunk 2]
```diff
@@ -105,23 +109,19 @@ To update or add new dependencies, run `go get <package name>`.
...
+15. goreleaser makes bad release notes... run the following to generate a better changelog: `git log --no-merges --pretty=format:'- %s %H (%aN)' ${PREVIOUS_RELEASE}..${RELEASE}`
     
 ### Some Comments
 
 There is a large number of steps here, mostly manual. Here are some comments and/or thoughts on it.
 
 * Step 3: Editing files for stable version. I can see that in one place, say, `README.md`. But do we really have to edit the code? Could we not use build-time flags as in `go build -ldflags="-X main.Version=foo"` or similar?
-* Step 4: use of personal tokens. We really should have a production token used to push releases out. We can discuss signing separately, but entering yours or my token in, to be used by Actions, and then removing it, is messy. It also means someone else could accidentally trigger with my token.
 * Step 5: This is the heart of the release. Adding a tag triggers a github release and an image build pushed to GHCR.
-* Step 5a: There are several steps in the GHCR push that probably could be simplfied down to a single one with the use of [docker/build-push-action@v2](https://github.com/marketplace/actions/build-and-push-docker-images)
```
Comment (qweeah) [inline docs/CLI/5_developer_guide.md:120]: Removed since we are already using buildx to push when build completes

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**